pkg/builder/builder.go

```diff
@@ -841,6 +860,161 @@ func makeScrutinizeInitContainers(vscr *v1beta1.VerticaScrutinize, vdb *vapi.Ver
+// BuildVProxyDeployment builds manifest for a subclusters VProxy deployment
+func BuildVProxyDeployment(nm types.NamespacedName, vdb *vapi.VerticaDB, sc *vapi.Subcluster) *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        nm.Name,
+			Namespace:   nm.Namespace,
+			Labels:      MakeLabelsForStsObject(vdb, sc),
+			Annotations: MakeAnnotationsForStsObject(vdb, sc),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: MakeDepSelectorLabels(vdb, sc),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      MakeLabelsForVProxyObject(vdb, sc, true),
+					Annotations: MakeAnnotationsForObject(vdb),
+				},
+				Spec: buildVProxyPodSpec(vdb, sc),
+			},
+			Replicas: &sc.Proxy.Replica,
+		},
+	}
+}
+
+// buildPodSpec creates a PodSpec for the deployment
+func buildVProxyPodSpec(vdb *vapi.VerticaDB, sc *vapi.Subcluster) corev1.PodSpec {
+	termGracePeriod := int64(vmeta.GetTerminationGracePeriodSeconds(vdb.Annotations))
+	return corev1.PodSpec{
+		NodeSelector:                  sc.NodeSelector,
+		Affinity:                      GetK8sAffinity(sc.Affinity),
+		Tolerations:                   sc.Tolerations,
+		ImagePullSecrets:              GetK8sLocalObjectReferenceArray(vdb.Spec.ImagePullSecrets),
+		Containers:                    []corev1.Container{makeVProxyContainer(vdb, sc)},
+		TerminationGracePeriodSeconds: &termGracePeriod,
+		ServiceAccountName:            vdb.Spec.ServiceAccountName,
+		SecurityContext:               vdb.Spec.PodSecurityContext,
+		Volumes: []corev1.Volume{
+			{
+				Name: sc.Name,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: sc.GetVProxyConfigMapName(vdb)},
+					},
+				},
+			},
+		},
+	}
+}
+
+// makeDataForProxyConfigMap generates a configmap data in config.yaml format.
+// # Vertica Proxy Configuration File Example
+//
+// # Proxy's listener network address for clients connections
+// listener:
+//
+//	host: ""  # Listen on all IP addresses of the local system
+//	port: 5433
+//
+// # The database for proxy connections
+// database:
+//
+//	nodes:
+//	  - node1.address.com:5433
+//	  - node2.address.com:5433
+//	  - node3.address.com:5433
+//
+// # Log level: 0=TRACE|1=DEBUG|2=INFO|3=WARN|4=FATAL|5=NONE (Default INFO)
+// log:
+//
+//	level: DEBUG
+//	# OR
+//	# level: 1
+//
+// tls:
+//
+//	# For proxy-server TLS
+//	serverca: /Path/to/server_cacert.pem
+//	hostname: db.example.com
+//	proxykey: /Path/to/proxy_key.pem
+//	proxycert: /Path/to/proxy_cert.pem
+//
+// # For client-proxy TLS
+// serverkey: /Path/to/server_key.pem
+// servercert: /Path/to/server_cert.pem
+// clientca: /Path/to/client_cacert.pem
+func makeDataForVProxyConfigMap(vdb *vapi.VerticaDB, sc *vapi.Subcluster) string {
+	var nodeList []string
+	port := 5433
+
+	for scIndex := int32(0); scIndex < sc.Size; scIndex++ {
```

Comment:
```suggestion
	for i := int32(0); i < sc.Size; i++ {
```
You are iterating over a single subcluster